libdnf/base/goal.cpp

```diff
@@ -565,9 +573,12 @@ void Goal::Impl::add_rpms_to_goal(base::Transaction & transaction) {
 
     rpm::PackageQuery installed(base, rpm::PackageQuery::ExcludeFlags::IGNORE_EXCLUDES);
     installed.filter_installed();
-    for (auto & [action, ids, settings] : rpm_ids) {
+    for (auto [action, ids, settings] : rpm_ids) {
```

Comment:
Perfect, please can you add a note that the iteration is performed on copies on purpose? I just want to prevent reverting the change due to performance optimization.